keps/0019-package-api-versioning.md

```diff
@@ -2,64 +2,374 @@
+- The progression of the package version is confusing with regards to the application version progression
+
+  Example:
+
+  | Time | Package/operator version | App version        | Comment                                            |
+  | ---- | ---------------          | ------------------ | -------------------------------------------------- |
+  | T0   | 1.0.0                    | 2.3.0              | Initial release based on app version 2.3.x         |
+  | T1   | 1.1.0                    | 3.0.0              | Release based on app version 3.0.x                 |
+  | T2   | 1.2.0                    | 2.3.0              | New operator-related feature for app version 2.3.x |
+  | T2   | 1.3.0                    | 3.0.0              | New operator-related feature for app version 3.0.x |
+  | T3   | 1.4.0                    | 2.3.1              | Bug fix release for app version 2.3.x              |
+  | T4   | 1.5.0                    | 2.3.1              | Important bug fix A in operator-related code       |
+  | T4   | 1.6.0                    | 3.0.0              | Important bug fix A in operator-related code       |
+  | T5   | 1.6.1                    | 2.3.1              | Small bug fix B in operator-related code           |
+  | T5   | 1.6.2                    | 3.0.0              | Small bug fix B in operator-related code           |
+
+  The example above illustrates two sources of confusion:
+  - A higher package version can contain a lower app version
+  - There's no relation between the package version growth and the app version growth
+
+#### Composite package/operator version
+
+##### Pros
+
+- Puts the app version (what the user cares about) front and center
+- Makes it trivial to have an "app version"-based backport/release strategy
+- Makes it trivial to have an "app version"-based support strategy (e.g., an organization communicates official support for only the N-2 versions of an operator)
+
+##### Cons
+
+- Visually more complex
+- Makes it complex to know which operator-related features are available in which versions
+
+##### Example hypothetical timeline for Apache Cassandra releases
+
+Remember that _version_ is _operator version_ is _package version_.
+
+| Time |      Version | App version | Operator revision | KUDO API version | Comment                                    | Operator revision change |
+| ---- | ------------ | ----------- | ----------------- | ---------------- | ------------------------------------------ | ------------------------ |
+| T0   | 3.11.4-0.1.0 |      3.11.4 |             0.1.0 | v1beta1          | Initial release based on Apache C\* 3.11.x | -                        |
```

Comment:
lets add an example of free-form string version also
`wheezy-0.1.0`
`sarge-0.1.0`
`stretch-0.1.0`
`sarge-0.2.0`